prediction_market_agent_tooling/tools/contract.py

```
@@ -909,84 +909,75 @@ def contract_implements_function(
         singleton_address = eth_abi.decode(["address"], raw_slot_0)[0]
-        # Recurse into singleton
-        return contract_implements_function(
-            Web3.to_checksum_address(singleton_address),
-            function_name=function_name,
-            web3=web3,
-            function_arg_types=function_arg_types,
-            look_for_proxy_contract=False,
-        )
+        return Web3.to_checksum_address(singleton_address)
     except DecodingError:
         logger.info(f"Error decoding contract address for singleton")
```

Ruff (0.12.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)